tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,515 @@
+        """
+        Test case to check get_entity_entity_hierarchy() for RegisteredEdgeSystem->RegisteredMetric
+        :return: None
+        """
+
+        # Register the edge
+        registered_entity = self.aws.register(self.edge_system)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = self.aws.register(test_metric)
+
+        # Creating the parent-child relationship
+        self.aws.create_relationship(registered_entity, registered_metric)
+
+        # Getting the parent child relationship array from registered metric
+        entity_hierarchy = self.aws._get_entity_hierarchy(registered_metric)
+
+        self.assertSequenceEqual([registered_entity.ref_entity.name, registered_metric.ref_entity.name],
+                                 entity_hierarchy, "Check the implementation of _get_entity_hierarchy for "
+                                                   "RegisteredEdgeSystem->RegisteredMetric")
+
+    def test_implementation_get_entity_hierarchy_device_metric(self):
+        """
+        Test case to check get_entity_entity_hierarchy() for RegisteredEdgeSystem->RegisteredDevice->RegisteredMetric
+        :return: None
+        """
+
+        # Register the edge
+        registered_entity = self.aws.register(self.edge_system)
+
+        #  Creating Simulated Device
+        test_sensor = SimulatedDevice("TestSensor")
+
+        #  Registering Device and creating Parent-Child relationship
+        reg_test_sensor = self.aws.register(test_sensor)
+
+        self.aws.create_relationship(registered_entity, reg_test_sensor)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=ureg.degC,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = self.aws.register(test_metric)
+
+        # Creating the parent-child relationship
+        self.aws.create_relationship(reg_test_sensor, registered_metric)
+
+        # Getting the parent child relationship array from registered metric
+        entity_hierarchy = self.aws._get_entity_hierarchy(registered_metric)
+
+        self.assertSequenceEqual([registered_entity.ref_entity.name, reg_test_sensor.ref_entity.name,
+                                  registered_metric.ref_entity.name],
+                                 entity_hierarchy, "Check the implementation of _get_entity_hierarchy for "
+                                                   "RegisteredEdgeSystem->RegisteredDevice->RegisteredMetric")
+
+    def test_validation_get_entity_hierarchy(self):
+        """
+        Test case to check the validation of _get_entity_hierarchy method for RegisteredMetric object.
```

Comment:
Please ensure that reviews are not missed.  Change comment to:
Test case to check the validation of _get_entity_hierarchy method for Metric object.